### PR TITLE
Add macos grpc_distribtests_python and grpc_distribtests_ruby

### PR DESCRIPTION
**Pull request by jtattermusch**
Add the "single-job" distribtests for python and ruby on macos. The main motivation is to prepare ground for running artifact build  & packages build & distribtests in a single-stage build (for which the green/red status is easily visible, unlike the 3 stage chained build artifact -> packages -> distribtests we currently have). This is to simplify the release process.

- somewhat counterintutively, I'm adding the "distribtest" jobs for python and ruby on macos despite the fact that these languages currently don't define any macos distribtests. This is mostly to make things consistent with other "single-job" distribtest jobs. Also, at some point the distribtests could easily be added and these new jobs would make that simple.
- by splitting the artifact & distribtest into per-languages, it will eventually be possible to get rid of the "macos/build_artifacts" job (which is one of the slowest macos jobs we have). The per-language jobs will be much faster  since each of them will do less (and will need to install fewer prerequisites).
- this PR also fixes some typos in already existing grpc_distribtest_* jobs.

### CODE REVIEW

**Comment by jtattermusch** [comment]
Passing distribtests_python job: https://source.cloud.google.com/results/invocations/77d2ef13-42af-4725-bca4-854d74c6dd57/targets (obtained by kidnapping an existing PR job).

File: tools/internal_ci/macos/grpc_build_artifacts.sh
Comment (lidizheng) [inline tools/internal_ci/macos/grpc_build_artifacts.sh:31]: Unrelated: do you think we should break this job into multiple ones to speed up?

File: tools/internal_ci/macos/grpc_build_artifacts.sh
Comment (lidizheng) [inline tools/internal_ci/macos/grpc_build_artifacts.sh:30]: 👍

File: tools/internal_ci/macos/pull_request/grpc_distribtests_php.cfg
Comment (lidizheng) [inline tools/internal_ci/macos/pull_request/grpc_distribtests_php.cfg:18]: s/ruby/php/?

File: tools/internal_ci/macos/grpc_distribtests_python.sh
Comment (lidizheng) [inline tools/internal_ci/macos/grpc_distribtests_python.sh:50]: Feel free to assign to me. Adding distribtests for Python on macOS is valuable.

File: tools/internal_ci/macos/pull_request/grpc_distribtests_php.cfg
Comment (jtattermusch) [reply tools/internal_ci/macos/pull_request/grpc_distribtests_php.cfg:18]: this is from the "DO NOT MERGE" commit (to kidnap the job and get and adhoc test result). I'll remove before merging.

File: tools/internal_ci/macos/grpc_distribtests_python.sh
Comment (jtattermusch) [reply tools/internal_ci/macos/grpc_distribtests_python.sh:50]: +1 Feel free to add the tests once this PR is in, I'll be happy to review. (note that "TODO(jtattermusch)" doesn't mean I'm signed up for the work, just that I added the TODO and have the context.)

File: tools/internal_ci/macos/grpc_build_artifacts.sh
Comment (jtattermusch) [reply tools/internal_ci/macos/grpc_build_artifacts.sh:31]: I think so and it's my plan eventually but it's going to be a multi-step process. 
(well, eventually this job is going to be obsolete, since we're going to have per-language jobs that do both building the artifacts and test them on the spot).

File: tools/internal_ci/linux/grpc_distribtests_php.sh
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_distribtests_php.sh:37]: kind of, for some languages the package phase is just "collect the artifacts from subdirectories" (so that the per-language release processes can expect the packages being found in the right place).
For some languages (like C# and ruby's grpc-tools), we will still need a second phase ("build packages") but that's more of an exception.

File: tools/internal_ci/macos/pull_request/grpc_distribtests_php.cfg
Comment (jtattermusch) [reply tools/internal_ci/macos/pull_request/grpc_distribtests_php.cfg]: removed.

**Comment by jtattermusch** [comment]
I added the ad-hoc test result link above and removed the DO NOT MERGE commit. This should now be ready to merge (once you LGTM).

**Review by lidizheng — APPROVED** [review]

**Pull request merged**
